aten/src/ATen/mps/EmptyTensor.cpp

```diff
@@ -65,13 +76,25 @@ TensorBase empty_strided_mps(
     IntArrayRef stride,
     ScalarType dtype,
     c10::optional<Device> device_opt) {
-  auto device = device_or_default(device_opt);
-  TORCH_INTERNAL_ASSERT(device.is_mps());
-  const DeviceGuard device_guard(device);
-  auto* allocator = at::mps::GetMPSAllocator();
-  constexpr c10::DispatchKeySet mps_dks(c10::DispatchKey::MPS);
-  return at::detail::empty_strided_generic(
-      size, stride, allocator, mps_dks, dtype);
+#if defined(__APPLE__)
+#if __is_target_os(macOS)
+  if (__builtin_available(macOSApplicationExtension 12.3, *)) {
+    auto device = device_or_default(device_opt);
+    TORCH_INTERNAL_ASSERT(device.is_mps());
+    const DeviceGuard device_guard(device);
+    auto* allocator = at::mps::GetMPSAllocator();
+    constexpr c10::DispatchKeySet mps_dks(c10::DispatchKey::MPS);
+    return at::detail::empty_strided_generic(
+        size, stride, allocator, mps_dks, dtype);
+  } else {
+    TORCH_INTERNAL_ASSERT(false, "MPS backend supported MacOS 12.3+")
```

Comment:
This is not an internal assert? This is the main assert that people are going to see if they try to create a Tensor on MPS device right?
If so, we should flesh out this message that the macos version you're running should be 12.3+.

Reply:
Done. I changed to TORCH_CHECK(). Currently I haven't included the information of the OS they are running on. It can be queried using `sw_vers`